File: practica1/P1S3/Carne.cpp
#pragma once
#include "Comida.cpp"
#include <iostream>
using namespace std;
class Carne : public Comida
{    
    public:
        Carne() :Comida()
        {  
            precio = 5.0;
        }

        double getPrecio()
        {
            return precio;
        }

        void setPrecio(double p)
        {
            precio = p;
        }

        string toString()
        {
            string resultado = "La carne cuesta " + to_string(precio);
            return resultado;
        }
};

File: practica1/P1S3/Java/src/practica1dss3/Carne.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practica1dss3;

/**
 *
 * @author USUARIO
 */
public class Carne implements Comida{

    private double precio;
    private boolean hayStock;
    
    public Carne(){
        precio = 5.0;
        hayStock = true;
    }
    
    @Override
    public double getPrecio() {
        return precio;
    }

    @Override
    public void setPrecio(double p) {
        precio = p;
    }
    
    @Override
    public void cambioStock(){
        hayStock = false;
    }
    
    @Override
    public String toString(){
        if(hayStock)
            return("La carne cuesta " + getPrecio());
        else
            return("Ya no queda carne");
    }
    
}
